#! /bin/sh
patch -p1 -l -f $* < $0
exit $?

--- a/CMakeLists.txt.orig 2022-08-11 16:55:07.647126624 +0200
+++ b/CMakeLists.txt      2022-08-16 17:31:31.609503511 +0200
@@ -26,7 +26,7 @@

 set(REQUIRED_QT_VERSION 5.15.2)
 find_package(Qt${QT_MAJOR_VERSION} ${REQUIRED_QT_VERSION} CONFIG REQUIRED Widgets)
-if (NOT ANDROID)
+if (NOT ANDROID AND NOT WIN32)
   find_package(Qt${QT_MAJOR_VERSION} ${REQUIRED_QT_VERSION} CONFIG REQUIRED DBus)
 endif()

@@ -44,7 +44,7 @@
 cmake_dependent_option(BUILD_DESIGNERPLUGIN "Build plugin for Qt Designer" ON "NOT CMAKE_CROSSCOMPILING" OFF)
 add_feature_info(DESIGNERPLUGIN ${BUILD_DESIGNERPLUGIN} "Build plugin for Qt Designer")

-if (NOT ANDROID AND NOT WIN32)
+if (FALSE)
     option(WITH_KAUTH "Build with KAuth support" ON)
 else()
     set(WITH_KAUTH OFF)
