#! /bin/sh
patch -p0 -l -f $* < $0
exit $?

--- CMakeLists.txt.orig 2022-12-09 10:14:20.972929616 +0100
+++ CMakeLists.txt      2022-12-09 10:51:29.381331575 +0100
@@ -1,7 +1,7 @@
 cmake_minimum_required(VERSION 3.16)

-set(KF_VERSION "5.99.0") # handled by release scripts
-set(KF_DEP_VERSION "5.99.0") # handled by release scripts
+set(KF_VERSION "5.100.0") # handled by release scripts
+set(KF_DEP_VERSION "5.100.0") # handled by release scripts
 project(KConfigWidgets VERSION ${KF_VERSION})

 include(FeatureSummary)
