#! /bin/sh
patch -p1 -l -f -R $* < $0
exit $?

diff --git a/src/conf/appearanceconfigpage.cpp b/src/conf/appearanceconfigpage.cpp
index 504f68ec..5284c246 100644
--- a/src/conf/appearanceconfigpage.cpp
+++ b/src/conf/appearanceconfigpage.cpp
@@ -31,7 +31,6 @@
 */

 #include <config-kleopatra.h>
-#include <kconfigwidgets_version.h>
 #include "appearanceconfigpage.h"

 #include "appearanceconfigwidget.h"
@@ -50,11 +49,7 @@ AppearanceConfigurationPage::AppearanceConfigurationPage(QWidget *parent, const
     mWidget = new AppearanceConfigWidget(this);
     lay->addWidget(mWidget);

-#if KCONFIGWIDGETS_VERSION < QT_VERSION_CHECK(5, 64, 0)
-    connect(mWidget, &AppearanceConfigWidget::changed, this, QOverload<>::of(&Kleo::Config::AppearanceConfigurationPage::changed));
-#else
     connect(mWidget, &AppearanceConfigWidget::changed, this, &Kleo::Config::AppearanceConfigurationPage::markAsChanged);
-#endif

     load();
 }
diff --git a/src/conf/cryptooperationsconfigpage.cpp b/src/conf/cryptooperationsconfigpage.cpp
index cdda69ee..d0c377c4 100644
--- a/src/conf/cryptooperationsconfigpage.cpp
+++ b/src/conf/cryptooperationsconfigpage.cpp
@@ -31,7 +31,6 @@
 */

 #include <config-kleopatra.h>
-#include <kconfigwidgets_version.h>

 #include "cryptooperationsconfigpage.h"

@@ -49,11 +48,7 @@ CryptoOperationsConfigurationPage::CryptoOperationsConfigurationPage(QWidget *pa
     lay->setContentsMargins(0, 0, 0, 0);
     mWidget = new CryptoOperationsConfigWidget(this);
     lay->addWidget(mWidget);
-#if KCONFIGWIDGETS_VERSION < QT_VERSION_CHECK(5, 64, 0)
-    connect(mWidget, &CryptoOperationsConfigWidget::changed, this, QOverload<>::of(&Kleo::Config::CryptoOperationsConfigurationPage::changed));
-#else
     connect(mWidget, &CryptoOperationsConfigWidget::changed, this, &Kleo::Config::CryptoOperationsConfigurationPage::markAsChanged);
-#endif
     load();
 }

diff --git a/src/conf/gnupgsystemconfigurationpage.cpp b/src/conf/gnupgsystemconfigurationpage.cpp
index 897b6507..fad1f45d 100644
--- a/src/conf/gnupgsystemconfigurationpage.cpp
+++ b/src/conf/gnupgsystemconfigurationpage.cpp
@@ -31,7 +31,6 @@
 */

 #include <config-kleopatra.h>
-#include <kconfigwidgets_version.h>
 #include "gnupgsystemconfigurationpage.h"

 #include <libkleo/cryptoconfigmodule.h>
@@ -56,11 +55,7 @@ GnuPGSystemConfigurationPage::GnuPGSystemConfigurationPage(QWidget *parent, cons
     lay->addWidget(mWidget);


-#if KCONFIGWIDGETS_VERSION < QT_VERSION_CHECK(5, 64, 0)
-    connect(mWidget, &CryptoConfigModule::changed, this, QOverload<>::of(&Kleo::Config::GnuPGSystemConfigurationPage::changed));
-#else
     connect(mWidget, &CryptoConfigModule::changed, this, &Kleo::Config::GnuPGSystemConfigurationPage::markAsChanged);
-#endif

     load();
 }
diff --git a/src/conf/smimevalidationconfigurationpage.cpp b/src/conf/smimevalidationconfigurationpage.cpp
index 12b1a26d..5476b1b3 100644
--- a/src/conf/smimevalidationconfigurationpage.cpp
+++ b/src/conf/smimevalidationconfigurationpage.cpp
@@ -31,7 +31,6 @@
 */

 #include <config-kleopatra.h>
-#include <kconfigwidgets_version.h>
 #include "smimevalidationconfigurationpage.h"
 #include "smimevalidationconfigurationwidget.h"

@@ -49,11 +48,7 @@ SMimeValidationConfigurationPage::SMimeValidationConfigurationPage(QWidget *pare
     mWidget = new SMimeValidationConfigurationWidget(this);
     lay->addWidget(mWidget);

-#if KCONFIGWIDGETS_VERSION < QT_VERSION_CHECK(5, 64, 0)
-    connect(mWidget, &SMimeValidationConfigurationWidget::changed, this, QOverload<>::of(&Kleo::Config::SMimeValidationConfigurationPage::changed));
-#else
     connect(mWidget, &SMimeValidationConfigurationWidget::changed, this, &Kleo::Config::SMimeValidationConfigurationPage::markAsChanged);
-#endif
     load();
 }
