#! /bin/sh
patch -p1 -l -f $* < $0
exit $?

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7223ab9..754088c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -6,8 +6,8 @@ option(FORCE_DISABLE_KCMUTILS "Force building Kleopatra without KCMUtils. Doing
 option(DISABLE_KWATCHGNUPG "Don't build the kwatchgnupg tool [default=OFF]" OFF)

 # Standalone build. Find / include everything neccessary.
-set(KF5_VERSION "5.28.0")
-set(KMIME_VERSION "5.4.40")
+set(KF5_VERSION "5.25.0")
+set(KMIME_VERSION "5.2")
 set(LIBKLEO_VERSION "5.4.40")
 set(QT_REQUIRED_VERSION "5.5.0")
