#! /bin/sh
patch -p1 -l -f $* < $0
exit $?

diff --git a/src/kleopatraapplication.cpp b/src/kleopatraapplication.cpp
index 0528076b..d4cb93ac 100644
--- a/src/kleopatraapplication.cpp
+++ b/src/kleopatraapplication.cpp
@@ -299,6 +299,7 @@ void KleopatraApplication::init()
 {
 #ifdef Q_OS_WIN
     QWindowsWindowFunctions::setWindowActivationBehavior(QWindowsWindowFunctions::AlwaysActivateWindow);
+    KConfig::setWindowsRegistryKey(QStringLiteral("SOFTWARE\\GNU\\Kleopatra"));
 #endif
     const auto blockedUrlSchemes = Settings{}.blockedUrlSchemes();
     for (const auto &scheme : blockedUrlSchemes) {
