#! /bin/sh
patch -p1 -f $* < $0
exit $?

diff -rup libetpan-0.54-orgi/tests/option-parser.c libetpan-0.54/tests/option-parser.c
--- libetpan-0.54-orgi/tests/option-parser.c	2008-02-20 23:15:53.000000000 +0100
+++ libetpan-0.54/tests/option-parser.c	2008-04-25 17:24:27.000000000 +0200
@@ -110,7 +110,7 @@ int parse_options(int argc, char ** argv
   * flags_directory = NULL;
 
   while (1) {
-    r = getopt_long(argc, argv, "d:s:p:txu:v:l:ac:f:D", long_options, &index);
+    r = getopt_long(argc, argv, "d:s:p:txu:v:l:ac:f:D", long_options, &indx);
     
     if (r == -1)
       break;
Only in libetpan-0.54/tests: option-parser.c~
