#! /bin/sh
patch -p1 -f $* < $0
exit $?

diff -rup libetpan-0.54/libetpan-config.h.in libetpan-0.52-dev-20071209/libetpan-config.h.in
--- libetpan-0.54/libetpan-config.h.in	2008-02-17 14:13:25.000000000 +0100
+++ libetpan-0.52-dev-20071209/libetpan-config.h.in	2007-12-09 00:57:43.000000000 +0100
@@ -6,7 +6,7 @@
 #define CONFIG_H
 #include "config.h"
 #endif
-@ifdef WIN32
+@if WIN32
 @	define MMAP_UNAVAILABLE
 @endif
 @ifdef _MSC_VER
