#! /bin/sh
patch -p1 -l -f $* < $0
exit $?

Fixes missing LDAP_OPT_SUCCESS.

--- orig/claws-mail-3.9.0/src/ldaputil.h	2012-05-27 19:16:13.000000000 +0200
+++ claws-mail-3.9.0/src/ldaputil.h	2013-05-02 19:58:31.275669432 +0200
@@ -14,7 +14,7 @@
  *
  * You should have received a copy of the GNU General Public License
  * along with this program. If not, see <http://www.gnu.org/licenses/>.
- *
+ *
  */

 /*
@@ -37,6 +37,9 @@
 #define ldap_unbind_ext(ld,x,y) ldap_unbind_s(ld)
 #define LDAP_ADMINLIMIT_EXCEEDED LDAP_ADMIN_LIMIT_EXCEEDED
 #define timeval l_timeval
+#ifndef LDAP_OPT_SUCCESS
+# define LDAP_OPT_SUCCESS 0
+#endif
 #endif
 /* Function Prototypes */
 GList *ldaputil_read_basedn	( const gchar *host, const gint port,
