#! /bin/sh
patch -p1 -f $* < $0
exit $?

Add missing symbols.

--- claws-mail-3.5.0cvs114/src/plugins/pgpcore/claws.def.~1.1.2.3.~	2008-08-01 20:28:37.000000000 +0200
+++ claws-mail-3.5.0cvs114/src/plugins/pgpcore/claws.def	2008-09-19 17:09:41.000000000 +0200
@@ -12,6 +12,30 @@
 extract_address
 get_rc_dir
 get_tmp_file
+gtk_cmclist_append
+gtk_cmclist_clear
+gtk_cmclist_get_row_data
+gtk_cmclist_get_type
+gtk_cmclist_get_type
+gtk_cmclist_get_type
+gtk_cmclist_get_type
+gtk_cmclist_get_type
+gtk_cmclist_get_type
+gtk_cmclist_get_type
+gtk_cmclist_get_type
+gtk_cmclist_get_type
+gtk_cmclist_new_with_titles
+gtk_cmclist_set_column_width
+gtk_cmclist_set_column_width
+gtk_cmclist_set_column_width
+gtk_cmclist_set_column_width
+gtk_cmclist_set_column_width
+gtk_cmclist_set_compare_func
+gtk_cmclist_set_compare_func
+gtk_cmclist_set_row_data_full
+gtk_cmclist_set_selection_mode
+gtk_cmclist_set_sort_type
+gtk_cmclist_sort
 gtkut_get_options_frame
 gtkut_stock_button_set_create
 gtkut_window_new
