#! /bin/sh
patch -p1 -l -f $* < $0
exit $?

diff --git a/src/kmessagewidget.cpp b/src/kmessagewidget.cpp
index b48395f8..d33d641f 100644
--- a/src/kmessagewidget.cpp
+++ b/src/kmessagewidget.cpp
@@ -109,6 +109,8 @@ void KMessageWidgetPrivate::createLayout()
         QToolButton *button = new QToolButton(q);
         button->setDefaultAction(action);
         button->setToolButtonStyle(Qt::ToolButtonTextBesideIcon);
+        auto previous = buttons.isEmpty() ? static_cast<QWidget *>(textLabel) : buttons.back();
+        QWidget::setTabOrder(previous, button);
         buttons.append(button);
     }

--
