#! /bin/sh
patch -p1 -l -f $* < $0
exit $?

diff --git a/sm/Makefile.in b/sm/Makefile.in
index b2f042de8..c676adac9 100644
--- a/sm/Makefile.in
+++ b/sm/Makefile.in
@@ -557,7 +557,7 @@ t_common_ldadd = $(libcommon) $(LIBGCRYPT_LIBS) $(KSBA_LIBS) \
                  $(GPG_ERROR_LIBS) $(LIBINTL) $(LIBICONV)

 t_minip12_CFLAGS = -DWITHOUT_NPTH=1 \
-	           $(LIBGCRYPT_CFLAGS) $(GPG_ERROR_CFLAGS)
+	           $(LIBGCRYPT_CFLAGS) $(KSBA_CFLAGS) $(GPG_ERROR_CFLAGS)

 t_minip12_SOURCES = $(t_common_src) t-minip12.c minip12.c
 t_minip12_LDADD   = $(t_common_ldadd) $(NETLIBS)
