#! /bin/sh
patch -p1 -l -f $* < $0
exit $?

diff --git a/dirmngr/Makefile.in b/dirmngr/Makefile.in
index f56e59a3e..d154e5228 100644
--- a/dirmngr/Makefile.in
+++ b/dirmngr/Makefile.in
@@ -790,7 +790,7 @@ t_ldap_parse_uri_SOURCES = \

 t_ldap_parse_uri_CFLAGS = -DWITHOUT_NPTH=1  $(USE_C99_CFLAGS) \
 			  $(LIBGCRYPT_CFLAGS) \
-                          $(LIBASSUAN_CFLAGS) $(GPG_ERROR_CFLAGS)
+                          $(LIBASSUAN_CFLAGS) $(GPG_ERROR_CFLAGS) $(KSBA_CFLAGS)

 t_ldap_parse_uri_LDADD = $(ldaplibs) $(t_common_ldadd) $(DNSLIBS)
 t_ldap_misc_SOURCES = t-ldap-misc.c ldap-misc.c ldap-misc.h $(ldap_url)
diff --git a/kbx/Makefile.in b/kbx/Makefile.in
index dd727e1e0..f5fe341bc 100644
--- a/kbx/Makefile.in
+++ b/kbx/Makefile.in
@@ -577,8 +577,8 @@ client_sources = \

 libkeybox_a_SOURCES = $(common_sources) $(client_sources)
 libkeybox509_a_SOURCES = $(common_sources) $(client_sources)
-libkeybox_a_CFLAGS = $(AM_CFLAGS) $(LIBASSUAN_CFLAGS)
-libkeybox509_a_CFLAGS = $(AM_CFLAGS) $(LIBASSUAN_CFLAGS) -DKEYBOX_WITH_X509=1
+libkeybox_a_CFLAGS = $(AM_CFLAGS) $(LIBASSUAN_CFLAGS) $(NPTH_CFLAGS)
+libkeybox509_a_CFLAGS = $(AM_CFLAGS) $(LIBASSUAN_CFLAGS) $(NPTH_CFLAGS) -DKEYBOX_WITH_X509=1

 # We need W32SOCKLIBS because the init subsystem code in libcommon
 # requires it - although we don't actually need it.  It is easier
diff --git a/tools/Makefile.in b/tools/Makefile.in
index 4c29f84ed..9e9e13eda 100644
--- a/tools/Makefile.in
+++ b/tools/Makefile.in
@@ -744,7 +744,7 @@ gpgtar_SOURCES = \
 	gpgtar-extract.c \
 	gpgtar-list.c

-gpgtar_CFLAGS = $(GPG_ERROR_CFLAGS)
+gpgtar_CFLAGS = $(LIBGCRYPT_CFLAGS) $(GPG_ERROR_CFLAGS)
 gpgtar_LDADD = $(libcommon) $(LIBGCRYPT_LIBS) $(GPG_ERROR_LIBS) \
                $(LIBINTL) $(NETLIBS) $(LIBICONV) $(W32SOCKLIBS) \
 	       $(gpgtar_rc_objs)
@@ -759,7 +759,7 @@ gpg_wks_server_SOURCES = \
 	mime-maker.c  mime-maker.h  \
 	send-mail.c   send-mail.h

-gpg_wks_server_CFLAGS = $(GPG_ERROR_CFLAGS) $(INCICONV)
+gpg_wks_server_CFLAGS = $(LIBGCRYPT_CFLAGS) $(GPG_ERROR_CFLAGS) $(INCICONV)
 gpg_wks_server_LDADD = $(libcommon) $(LIBGCRYPT_LIBS) $(GPG_ERROR_LIBS) \
 		       $(LIBINTL) $(LIBICONV)

@@ -774,7 +774,8 @@ gpg_wks_client_SOURCES = \
 	send-mail.c   send-mail.h   \
 	call-dirmngr.c call-dirmngr.h

-gpg_wks_client_CFLAGS = $(LIBASSUAN_CFLAGS) $(GPG_ERROR_CFLAGS) $(INCICONV)
+gpg_wks_client_CFLAGS = $(LIBASSUAN_CFLAGS) $(LIBGCRYPT_CFLAGS) \
+			$(GPG_ERROR_CFLAGS) $(INCICONV)
 gpg_wks_client_LDADD = $(libcommon) \
 		       $(LIBASSUAN_LIBS) $(LIBGCRYPT_LIBS) $(GPG_ERROR_LIBS) \
 		       $(LIBINTL) $(LIBICONV) $(NETLIBS) \
@@ -783,7 +784,7 @@ gpg_wks_client_LDADD = $(libcommon) \
 gpg_pair_tool_SOURCES = \
 	gpg-pair-tool.c

-gpg_pair_tool_CFLAGS = $(GPG_ERROR_CFLAGS) $(INCICONV)
+gpg_pair_tool_CFLAGS = $(LIBGCRYPT_CFLAGS) $(GPG_ERROR_CFLAGS) $(INCICONV)
 gpg_pair_tool_LDADD =  $(libcommon) \
 		       $(LIBGCRYPT_LIBS) $(GPG_ERROR_LIBS) \
 		       $(LIBINTL) $(LIBICONV) $(W32SOCKLIBS)
