#! /bin/sh
patch -p0 -l -f $* < $0
exit $?

--- src/dirmngr.c.orig	2014-08-13 13:29:03.609026165 +0000
+++ src/dirmngr.c	2014-08-13 13:30:19.328411198 +0000
@@ -668,12 +668,14 @@
      Note that this will also do the pth_init. */

   /* Init Libgcrypt. */
+#if GCRYPT_VERSION_NUMBER < 0x010600
   rc = gcry_control (GCRYCTL_SET_THREAD_CBS, &gcry_threads_pth);
   if (rc)
     {
       log_fatal ("can't register GNU Pth with Libgcrypt: %s\n",
                  gpg_strerror (rc));
     }
+#endif
   gcry_control (GCRYCTL_DISABLE_SECMEM, 0);
   if (!gcry_check_version (NEED_LIBGCRYPT_VERSION) )
     {
