#! /bin/sh
patch -p1 -l -f $* < $0
exit $?

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7223ab9..754088c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2,7 +2,7 @@ cmake_minimum_required(VERSION 2.8.12)

 project(libkleo)

-set(KF5_VERSION "5.30.0")
+set(KF5_VERSION "5.25.0")
 find_package(ECM ${KF5_VERSION} CONFIG REQUIRED)
 set(CMAKE_MODULE_PATH ${ECM_MODULE_PATH})
 set(LIBRARY_NAMELINK)
