#! /bin/sh
patch -p1 -l -f $* < $0
exit $?

--- a/src/plugins/platforms/windows/qwindowstheme.cpp
+++ b/src/plugins/platforms/windows/qwindowstheme.cpp
@@ -1257,6 +1257,8 @@

 Qt::ColorScheme QWindowsTheme::queryColorScheme()
 {
+    if (QOperatingSystemVersion::current() < QOperatingSystemVersion::Windows10_1809)
+        return Qt::ColorScheme::Unknown;
     if (queryHighContrast())
         return Qt::ColorScheme::Unknown;
