i#! /bin/sh
patch -p1 -f -l $* < $0
exit $?

diff -Nru qtbase-5.6.1.orig/src/corelib/io/qstandardpaths.cpp qtbase-5.6.1/src/corelib/io/qstandardpaths.cpp
--- qtbase-5.6.1.orig/src/corelib/io/qstandardpaths.cpp 2016-07-22 15:11:23.307466618 +0200
+++ qtbase-5.6.1/src/corelib/io/qstandardpaths.cpp      2016-07-22 15:12:52.629403686 +0200
@@ -56,6 +56,14 @@
 #include <unistd.h>
 #endif

+#if !defined(QT_BUILD_QMAKE) && !defined(QT_NO_SETTINGS)  && !defined(QT_BOOTSTRAPPED)
+#include <qsettings.h>
+#include <qcoreapplication.h>
+#endif
+#ifdef Q_OS_WIN
+#include <windows.h>
+#endif
+
 #ifndef QT_NO_STANDARDPATHS

 QT_BEGIN_NAMESPACE
@@ -672,6 +680,70 @@
     return qsp_testMode;
 }

+#if !defined(QT_BUILD_QMAKE) && !defined(QT_NO_SETTINGS) && !defined(QT_BOOTSTRAPPED)
+static const char standardPathsSection[] = "StandardPaths";
+static QSettings *findConfiguration()
+{
+    QString qtconfig = QStringLiteral(":/qt/etc/qt.conf");
+    if (QFile::exists(qtconfig))
+        return new QSettings(qtconfig, QSettings::IniFormat);
+    if (QCoreApplication::instance()) {
+        QDir pwd(QCoreApplication::applicationDirPath());
+        qtconfig = pwd.filePath(QLatin1String("qt.conf"));
+        if (QFile::exists(qtconfig))
+            return new QSettings(qtconfig, QSettings::IniFormat);
+    }
+    return 0;     //no luck
+}
+#endif
+
+QString QStandardPaths::presetLocation(StandardLocation type)
+{
+#if !defined(QT_BUILD_QMAKE) && !defined(QT_NO_SETTINGS) && !defined(QT_BOOTSTRAPPED)
+    QStringList locations = QStringList()   << QStringLiteral("DesktopLocation")
+                                            << QStringLiteral("DocumentsLocation")
+                                            << QStringLiteral("FontsLocation")
+                                            << QStringLiteral("ApplicationsLocation")
+                                            << QStringLiteral("MusicLocation")
+                                            << QStringLiteral("MoviesLocation")
+                                            << QStringLiteral("PicturesLocation")
+                                            << QStringLiteral("TempLocation")
+                                            << QStringLiteral("HomeLocation")
+                                            << QStringLiteral("DataLocation")
+                                            << QStringLiteral("CacheLocation")
+                                            << QStringLiteral("GenericDataLocation")
+                                            << QStringLiteral("RuntimeLocation")
+                                            << QStringLiteral("ConfigLocation")
+                                            << QStringLiteral("DownloadLocation")
+                                            << QStringLiteral("GenericCacheLocation")
+                                            << QStringLiteral("GenericConfigLocation")
+                                            << QStringLiteral("AppDataLocation")
+                                            << QStringLiteral("AppConfigLocation")
+                                            << QStringLiteral("AppLocalDataLocation");
+
+    QScopedPointer<const QSettings> settings(findConfiguration());
+    if (!settings.isNull()) {
+        QString key = QLatin1String(standardPathsSection);
+        key += QLatin1Char('/');
+        key += locations[type];
+        const QString value = settings->value(key).toString();
+#ifdef Q_OS_WIN
+        if (value.isEmpty())
+#endif
+          return value;
+#ifdef Q_OS_WIN
+        wchar_t expandedPath[MAX_PATH] = {0};
+        if (ExpandEnvironmentStrings((wchar_t*)value.utf16(), expandedPath, MAX_PATH)) {
+            return QString::fromWCharArray(expandedPath).replace("\\", "/");
+        }
+        return value;
+#endif
+    }
+#endif // !QT_BUILD_QMAKE && !QT_NO_SETTINGS
+
+    return QString();
+}
+

 QT_END_NAMESPACE

:diff -Nru qtbase-5.6.1.orig/src/corelib/io/qstandardpaths.h qtbase-5.6.1/src/corelib/io/qstandardpaths.h
--- qtbase-5.6.1.orig/src/corelib/io/qstandardpaths.h   2016-07-22 15:11:23.307466618 +0200
+++ qtbase-5.6.1/src/corelib/io/qstandardpaths.h        2016-07-22 15:11:49.940044214 +0200
@@ -92,6 +92,8 @@
     static bool isTestModeEnabled();

 private:
+    static QString presetLocation(StandardLocation type);
+
     // prevent construction
     QStandardPaths();
     ~QStandardPaths();
diff -Nru qtbase-5.6.1.orig/src/corelib/io/qstandardpaths_win.cpp qtbase-5.6.1/src/corelib/io/qstandardpaths_win.cpp
--- qtbase-5.6.1.orig/src/corelib/io/qstandardpaths_win.cpp     2016-07-22 15:11:23.307466618 +0200
+++ qtbase-5.6.1/src/corelib/io/qstandardpaths_win.cpp  2016-07-22 15:13:41.682467339 +0200
@@ -203,7 +203,9 @@

 QString QStandardPaths::writableLocation(StandardLocation type)
 {
-    QString result;
+    QString result = QStandardPaths::presetLocation(type);
+    if (!result.isEmpty())
+        return result;
     switch (type) {
     case DownloadLocation:
         result = sHGetKnownFolderPath(qCLSID_FOLDERID_Downloads, type);
