#! /bin/sh
patch -p1 -f -l $* < $0
exit $?

diff --git a/src/plugins/styles/windowsvista/qwindowsvistastyle.cpp b/src/plugins/styles/windowsvista/qwindowsvistastyle.cpp
index 140daeaf6d..de714914f3 100644
--- a/src/plugins/styles/windowsvista/qwindowsvistastyle.cpp
+++ b/src/plugins/styles/windowsvista/qwindowsvistastyle.cpp
@@ -1453,12 +1453,12 @@ void QWindowsVistaStyle::drawControl(ControlElement element, const QStyleOption
                 unsigned int res = GetThemeColor(theme.handle(), LVP_LISTITEM, LISS_SELECTED, TMT_TEXTCOLOR, &cref);
                 QColor textColor(GetRValue(cref), GetGValue(cref), GetBValue(cref));
                 */
-                QPalette palette = vopt->palette;
-                palette.setColor(QPalette::All, QPalette::HighlightedText, palette.color(QPalette::Active, QPalette::Text));
+                // QPalette palette = vopt->palette;
+                // palette.setColor(QPalette::All, QPalette::HighlightedText, palette.color(QPalette::Active, QPalette::Text));
                 // Note that setting a saturated color here results in ugly XOR colors in the focus rect
-                palette.setColor(QPalette::All, QPalette::Highlight, palette.base().color().darker(108));
+                // palette.setColor(QPalette::All, QPalette::Highlight, palette.base().color().darker(108));
                 QStyleOptionViewItem adjustedOption = *vopt;
-                adjustedOption.palette = palette;
+                // adjustedOption.palette = palette;
                 // We hide the  focusrect in singleselection as it is not required
                 if ((view->selectionMode() == QAbstractItemView::SingleSelection)
                     && !(vopt->state & State_KeyboardFocusChange))
