#! /bin/sh
patch -p1 -f -l $* < $0
exit $?

diff --git a/src/plugins/styles/windowsvista/qwindowsvistastyle.cpp b/src/plugins/styles/windowsvista/qwindowsvistastyle.cpp
index 140daeaf6d..a5bc282527 100644
--- a/src/plugins/styles/windowsvista/qwindowsvistastyle.cpp
+++ b/src/plugins/styles/windowsvista/qwindowsvistastyle.cpp
@@ -166,12 +166,22 @@ void deleteClonedAnimationStyleOption(const QStyleOption *option)
   \sa QMacStyle, QWindowsXPStyle, QFusionStyle
 */

+static bool queryHighContrast()
+{
+    HIGHCONTRAST hcf = {};
+    hcf.cbSize = static_cast<UINT>(sizeof(HIGHCONTRAST));
+    if (SystemParametersInfo(SPI_GETHIGHCONTRAST, hcf.cbSize, &hcf, FALSE))
+        return hcf.dwFlags & HCF_HIGHCONTRASTON;
+    return false;
+}
+
 /*!
   Constructs a QWindowsVistaStyle object.
 */
 QWindowsVistaStyle::QWindowsVistaStyle()
     : QWindowsXPStyle(*new QWindowsVistaStylePrivate)
 {
+    highContrastTheme = queryHighContrast();
 }

 /*!
@@ -1458,7 +1468,8 @@ void QWindowsVistaStyle::drawControl(ControlElement element, const QStyleOption
                 // Note that setting a saturated color here results in ugly XOR colors in the focus rect
                 palette.setColor(QPalette::All, QPalette::Highlight, palette.base().color().darker(108));
                 QStyleOptionViewItem adjustedOption = *vopt;
-                adjustedOption.palette = palette;
+                if (!highContrastTheme)
+                    adjustedOption.palette = palette;
                 // We hide the  focusrect in singleselection as it is not required
                 if ((view->selectionMode() == QAbstractItemView::SingleSelection)
                     && !(vopt->state & State_KeyboardFocusChange))
diff --git a/src/plugins/styles/windowsvista/qwindowsvistastyle_p.h b/src/plugins/styles/windowsvista/qwindowsvistastyle_p.h
index 43a2a670f8..fe40990147 100644
--- a/src/plugins/styles/windowsvista/qwindowsvistastyle_p.h
+++ b/src/plugins/styles/windowsvista/qwindowsvistastyle_p.h
@@ -102,6 +102,8 @@ private:
     Q_DISABLE_COPY_MOVE(QWindowsVistaStyle)
     Q_DECLARE_PRIVATE(QWindowsVistaStyle)
     friend class QStyleFactory;
+
+    bool highContrastTheme = false;
 };

 QT_END_NAMESPACE
