#! /bin/sh
patch -p0 -f $* < $0
exit $?

--- mkspecs/win32-g++/qmake.conf.orig   2016-01-05 17:15:29.915892960 +0100
+++ mkspecs/win32-g++/qmake.conf        2016-01-05 17:17:07.041560381 +0100
@@ -85,8 +85,8 @@

 QMAKE_LIBS              =
 QMAKE_LIBS_CORE         = -lole32 -luuid -lws2_32 -ladvapi32 -lshell32 -luser32 -lkernel32
-QMAKE_LIBS_GUI          = -lgdi32 -lcomdlg32 -loleaut32 -limm32 -lwinmm -lws2_32 -lole32 -luuid -luser32 -ladvapi32
-QMAKE_LIBS_NETWORK      = -lws2_32
+QMAKE_LIBS_GUI          = -lgdi32 -lcomdlg32 -loleaut32 -limm32 -lwinmm -lws2_32 -lole32 -luuid -luser32 -ladvapi32 -lz
+QMAKE_LIBS_NETWORK      = -lws2_32 -lz
 QMAKE_LIBS_OPENGL       = -lglu32 -lopengl32 -lgdi32 -luser32
 QMAKE_LIBS_OPENGL_ES2   = -llibEGL -llibGLESv2 -lgdi32 -luser32
 QMAKE_LIBS_OPENGL_ES2_DEBUG = -llibEGLd -llibGLESv2d -lgdi32 -luser32
