#! /bin/sh
patch -p1 -f -l $* < $0
exit $?

diff --git a/src/plugins/styles/windowsvista/qwindowsvistastyle.cpp b/src/plugins/styles/windowsvista/qwindowsvistastyle.cpp
index 140daeaf6d..80fa3c263e 100644
--- a/src/plugins/styles/windowsvista/qwindowsvistastyle.cpp
+++ b/src/plugins/styles/windowsvista/qwindowsvistastyle.cpp
@@ -2325,14 +2336,14 @@ void QWindowsVistaStyle::polish(QWidget *widget)
         //note that since tooltips are not reused
         //we do not have to care about unpolishing
         widget->setContentsMargins(3, 0, 4, 0);
-        COLORREF bgRef;
-        HTHEME theme = OpenThemeData(widget ? QWindowsVistaStylePrivate::winId(widget) : nullptr, L"TOOLTIP");
-        if (theme && SUCCEEDED(GetThemeColor(theme, TTP_STANDARD, TTSS_NORMAL, TMT_TEXTCOLOR, &bgRef))) {
-            QColor textColor = QColor::fromRgb(bgRef);
-            QPalette pal;
-            pal.setColor(QPalette::All, QPalette::ToolTipText, textColor);
-            widget->setPalette(pal);
-        }
+        // COLORREF bgRef;
+        // HTHEME theme = OpenThemeData(widget ? QWindowsVistaStylePrivate::winId(widget) : nullptr, L"TOOLTIP");
+        // if (theme && SUCCEEDED(GetThemeColor(theme, TTP_STANDARD, TTSS_NORMAL, TMT_TEXTCOLOR, &bgRef))) {
+        //     QColor textColor = QColor::fromRgb(bgRef);
+        //     QPalette pal;
+        //     pal.setColor(QPalette::All, QPalette::ToolTipText, textColor);
+        //     widget->setPalette(pal);
+        // }
     } else if (qobject_cast<QMessageBox *> (widget)) {
         widget->setAttribute(Qt::WA_StyledBackground);
 #if QT_CONFIG(dialogbuttonbox)
