#! /bin/sh
patch -p0 -R -f -l $* < $0
exit $?

--- src/plugins/platforms/windows/qwindowstheme.cpp     2017-06-12 11:35:29.491363130 +0200
+++ src/plugins/platforms/windows/qwindowstheme.cpp.orig        2017-06-12 11:35:24.223373397 +0200
@@ -623,7 +623,7 @@
         memset(&iconInfo, 0, sizeof(iconInfo));
         iconInfo.cbSize = sizeof(iconInfo);
         stockFlags |= (pixmapSize.width() > 16 ? SHGFI_LARGEICON : SHGFI_SMALLICON);
-        if (QWindowsContext::shell32dll.sHGetStockIconInfo(stockId, SHGFI_ICON | stockFlags, &iconInfo) == S_OK) {
+        if (SHGetStockIconInfo(stockId, SHGFI_ICON | stockFlags, &iconInfo) == S_OK) {
             pixmap = qt_pixmapFromWinHICON(iconInfo.hIcon);
             DestroyIcon(iconInfo.hIcon);
             return pixmap;
--- src/plugins/platforms/windows/qwindowswindow.cpp    2017-06-12 11:40:03.530833862 +0200
+++ src/plugins/platforms/windows/qwindowswindow.cpp.orig   2017-06-12 11:37:38.367113636 +0200
@@ -1599,7 +1599,8 @@

 static inline bool dwmIsCompositionEnabled()
 {
-    return false;
+    BOOL dWmCompositionEnabled = FALSE;
+    return SUCCEEDED(DwmIsCompositionEnabled(&dWmCompositionEnabled)) && dWmCompositionEnabled == TRUE;
 }

 bool QWindowsWindow::handleWmPaint(HWND hwnd, UINT message,
