#! /bin/sh
patch -p0 -f $* < $0
exit $?

Fix for 0.10.13


--- src/mimeparser.c	(revision 250)
+++ src/mimeparser.c	(working copy)
@@ -733,7 +733,7 @@
       if (s)
         filename = xstrdup (s);
       s = rfc822parse_query_parameter (field, NULL, 1);
-      if (s && !strcmp (s, "inline"))
+      if (s && strcmp (s, "inline"))
         not_inline_text = 1;
       rfc822parse_release_field (field);
     }




