From 2c21c1a3304ccfceb99d59d522e969bed005ef05 Mon Sep 17 00:00:00 2001
From: Andre Heinecke <aheinecke@gnupg.org>
Date: Wed, 10 Apr 2024 12:42:49 +0200
Subject: [PATCH] Do not fail if WITH_K7ZIP is not defined

---
 generators/comicbook/document.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/generators/comicbook/document.cpp b/generators/comicbook/document.cpp
index e31999fa2..308a26eb6 100644
--- a/generators/comicbook/document.cpp
+++ b/generators/comicbook/document.cpp
@@ -16,7 +16,7 @@
 #include <QMimeDatabase>
 #include <QMimeType>
 #include <kzip.h>
-#if WITH_K7ZIP
+#ifdef WITH_K7ZIP
 #include <K7Zip>
 #endif

--
2.44.0
