#! /bin/sh
patch -p1 -l -f $* < $0
exit $?

diff -pruN a/generators/poppler/formfields.cpp b/generators/poppler/formfields.cpp
--- a/generators/poppler/formfields.cpp	2026-06-10 12:06:00.578671658 +0200
+++ b/generators/poppler/formfields.cpp	2023-11-22 17:09:51.000000000 +0100
@@ -403,7 +403,7 @@ PopplerFormFieldSignature::PopplerFormFi
     m_info = fromPoppler(result.first);
     m_asyncObject = result.second;
     QObject::connect(m_asyncObject.get(), &Poppler::AsyncObject::done, m_asyncObject.get(), [this]() {
-        m_info = fromPoppler(m_field->validateResult());
+        m_info.setCertificateStatus(fromPoppler(m_field->validateResult()));
         for (auto [_, callback] : m_updateSubscriptions) {
             callback();
         }
diff -pruN a/generators/poppler/pdfsignatureutils.h b/generators/poppler/pdfsignatureutils.h
--- a/generators/poppler/pdfsignatureutils.h	2026-06-10 12:05:39.850801355 +0200
+++ b/generators/poppler/pdfsignatureutils.h	2023-11-22 17:09:51.000000000 +0100
@@ -15,6 +15,7 @@

 Okular::SignatureInfo fromPoppler(const Poppler::SignatureValidationInfo &info);
 Okular::CertificateInfo fromPoppler(const Poppler::CertificateInfo &info);
+Okular::SignatureInfo::CertificateStatus fromPoppler(Poppler::SignatureValidationInfo::CertificateStatus status);

 class PopplerCertificateStore : public Okular::CertificateStore
 {
