#! /bin/sh
patch -p0 -R -f $* < $0
exit $?

--- CMakeLists.txt	2016-01-21 13:38:48.962678469 +0100
+++ CMakeLists.txt.orig	2016-01-21 13:40:08.468110980 +0100
@@ -58,9 +58,7 @@
     ki18n_install(po)
 endif()
 add_subdirectory(src)
-if (BUILD_TESTING)
-  add_subdirectory(autotests)
-endif()
+add_subdirectory(autotests)

 # create a Config.cmake and a ConfigVersion.cmake file and install them
 set(CMAKECONFIG_INSTALL_DIR "${KDE_INSTALL_CMAKEPACKAGEDIR}/KF5I18n")
