#! /bin/sh
patch -p1 -l -f $* < $0
exit $?


--- a/src/common/plugin.c	2013-05-06 17:12:52.454020853 +0200
+++ b/src/common/plugin.c	2013-05-06 17:09:00.954027453 +0200
@@ -604,13 +604,11 @@
 void plugin_load_standard_plugins (void)
 {
	static const char *names[] = {
-#ifdef G_OS_WIN32
                "pgpmime",
                "pgpinline",
-#else
+                "smime",
                 /* post-2.5 maybe
                 "bogofilter", */
-#endif
                 NULL
        };
	int i;
