#! /bin/sh
patch -p1 -l -f $* < $0
exit $?

diff --git a/src/browserrun.cpp b/src/browserrun.cpp
index ae9dd0f..de6e87a 100644
--- a/src/browserrun.cpp
+++ b/src/browserrun.cpp
@@ -410,7 +410,9 @@ void KParts::BrowserRun::saveUrl(const QUrl &url, const QString &suggestedFileNa

                 auto *job = new KIO::CommandLauncherJob(downloadManager, args);
                 job->setExecutable(downloadManager);
+#if 0
                 job->setUiDelegate(new KDialogJobUiDelegate(KJobUiDelegate::AutoHandlingEnabled, window));
+#endif
                 job->start();
                 return;
             }
@@ -527,7 +529,9 @@ void BrowserRun::slotCopyToTempFileResult(KJob *job)
         // Same as KRun::foundMimeType but with a different URL
         const QUrl destUrl = static_cast<KIO::FileCopyJob *>(job)->destUrl();
         KIO::OpenUrlJob *job = new KIO::OpenUrlJob(destUrl, d->m_mimeType);
+#if 0
         job->setUiDelegate(new KIO::JobUiDelegate(KJobUiDelegate::AutoHandlingEnabled, d->m_window));
+#endif
         job->setRunExecutables(true);
         job->start();
     }
