#! /bin/sh
patch -p0 -f $* < $0
exit $?

--- src/mimeview.c	6 Jan 2009 16:45:59 -0000	1.83.2.151
+++ src/mimeview.c	8 Jan 2009 17:21:57 -0000	1.83.2.152
@@ -1451,13 +1451,13 @@
 		KEY_PRESS_EVENT_STOP();
 		mimeview_launch(mimeview, NULL);
 		return TRUE;
-#ifndef G_OS_WIN32
 	case GDK_o:
 		BREAK_ON_MODIFIER_KEY();
 		KEY_PRESS_EVENT_STOP();
+#ifndef G_OS_WIN32
 		mimeview_open_with(mimeview);
-		return TRUE;
 #endif
+		return TRUE;
 	case GDK_c:
 		BREAK_ON_MODIFIER_KEY();
 		KEY_PRESS_EVENT_STOP();

