#! /bin/sh
patch -p0 -f $* < $0
exit $?

Index: src/prefs_themes.c
===================================================================
RCS file: /srv/cvs/claws-mail/claws/src/prefs_themes.c,v
retrieving revision 1.3.2.61
retrieving revision 1.3.2.62
diff -u -r1.3.2.61 -r1.3.2.62
--- src/prefs_themes.c	2 Jan 2009 10:51:34 -0000	1.3.2.61
+++ src/prefs_themes.c	9 Jan 2009 08:02:05 -0000	1.3.2.62
@@ -253,14 +253,19 @@
 static gboolean prefs_themes_is_system_theme(const gchar *dirname)
 {
 	gint len;
-	
+	gchar *system_theme_dir;
+	gboolean is_sys = FALSE;
+
 	g_return_val_if_fail(dirname != NULL, FALSE);
 
-	len = strlen(PACKAGE_DATA_DIR);
-	if (strlen(dirname) > len && 0 == strncmp(dirname, PACKAGE_DATA_DIR, len))
-		return TRUE;
+	system_theme_dir = stock_pixmap_get_system_theme_dir_for_theme(NULL);
+	len = strlen(system_theme_dir);
+	if (strlen(dirname) > len && 0 == strncmp(dirname, system_theme_dir, len))
+		is_sys = TRUE;
 	
-	return FALSE;
+	g_free(system_theme_dir);
+
+	return is_sys;
 }
 
 static void prefs_themes_set_themes_menu(GtkComboBox *combo, const ThemesData *tdata)
@@ -526,9 +531,8 @@
 				 GTK_STOCK_NO, GTK_STOCK_YES, NULL);
 		switch (val) {
 		case G_ALERTALTERNATE:
-			cinfo->dest = g_strconcat(PACKAGE_DATA_DIR, G_DIR_SEPARATOR_S,
-						  PIXMAP_THEME_DIR, G_DIR_SEPARATOR_S, 
-						  themename, NULL);
+			cinfo->dest = stock_pixmap_get_system_theme_dir_for_theme(
+						themename);
 			break;
 		case G_ALERTDEFAULT:
 			break;
Index: src/stock_pixmap.c
===================================================================
RCS file: /srv/cvs/claws-mail/claws/src/stock_pixmap.c,v
retrieving revision 1.25.2.62
retrieving revision 1.25.2.63
diff -u -r1.25.2.62 -r1.25.2.63
--- src/stock_pixmap.c	2 Jan 2009 13:05:54 -0000	1.25.2.62
+++ src/stock_pixmap.c	9 Jan 2009 08:02:05 -0000	1.25.2.63
@@ -527,6 +527,19 @@
 	closedir(dp);
 }
 
+gchar *stock_pixmap_get_system_theme_dir_for_theme(const gchar *theme)
+{
+	const gchar *sep = NULL;
+	if (theme && *theme)
+		sep = G_DIR_SEPARATOR_S;
+#ifndef G_OS_WIN32
+	return g_strconcat(PACKAGE_DATA_DIR, G_DIR_SEPARATOR_S,
+	        	   PIXMAP_THEME_DIR, sep, theme, NULL);
+#else
+	return g_strconcat(get_themes_dir(), sep, theme, NULL);
+#endif
+}
+
 GList *stock_pixmap_themes_list_new(void)
 {
 	gchar *defaulttheme;
@@ -537,11 +550,9 @@
 	defaulttheme = g_strdup(DEFAULT_PIXMAP_THEME);
 	userthemes   = g_strconcat(get_rc_dir(), G_DIR_SEPARATOR_S, 
 				   PIXMAP_THEME_DIR, NULL);
-	systemthemes = g_strconcat(PACKAGE_DATA_DIR, G_DIR_SEPARATOR_S,
-				   PIXMAP_THEME_DIR, NULL);
+	systemthemes = stock_pixmap_get_system_theme_dir_for_theme(NULL);
 
 	list = g_list_append(list, defaulttheme);
-	
 	stock_pixmap_find_themes_in_dir(&list, userthemes);
 	stock_pixmap_find_themes_in_dir(&list, systemthemes);
 
Index: src/stock_pixmap.h
===================================================================
RCS file: /srv/cvs/claws-mail/claws/src/stock_pixmap.h,v
retrieving revision 1.18.2.39
retrieving revision 1.18.2.40
diff -u -r1.18.2.39 -r1.18.2.40
--- src/stock_pixmap.h	2 Jan 2009 13:05:54 -0000	1.18.2.39
+++ src/stock_pixmap.h	9 Jan 2009 08:02:05 -0000	1.18.2.40
@@ -216,5 +216,6 @@
 					     OverlayPosition	 pos,
 					     gint		 border_x,
 					     gint		 border_y);
+gchar *stock_pixmap_get_system_theme_dir_for_theme(const gchar *theme);
 
 #endif /* __STOCK_PIXMAP_H__ */
Index: src/common/utils.c
===================================================================
RCS file: /srv/cvs/claws-mail/claws/src/common/utils.c,v
retrieving revision 1.36.2.160
retrieving revision 1.36.2.161
diff -u -r1.36.2.160 -r1.36.2.161
--- src/common/utils.c	6 Jan 2009 09:26:48 -0000	1.36.2.160
+++ src/common/utils.c	9 Jan 2009 08:02:05 -0000	1.36.2.161
@@ -1996,6 +1996,21 @@
 #endif
 }
 
+
+#ifdef G_OS_WIN32
+/* Return the default directory for Themes. */
+const gchar *get_themes_dir(void)
+{
+	static gchar *themes_dir = NULL;
+
+	if (!themes_dir)
+		themes_dir = g_strconcat(w32_get_module_dir(),
+					 "\\share\\claws-mail\\themes",
+					 NULL);
+	return themes_dir;
+}
+#endif
+
 const gchar *get_tmp_dir(void)
 {
 	static gchar *tmp_dir = NULL;
Index: src/common/utils.h
===================================================================
RCS file: /srv/cvs/claws-mail/claws/src/common/utils.h,v
retrieving revision 1.20.2.65
retrieving revision 1.20.2.66
diff -u -r1.20.2.65 -r1.20.2.66
--- src/common/utils.h	13 Dec 2008 21:20:39 -0000	1.20.2.65
+++ src/common/utils.h	9 Jan 2009 08:02:05 -0000	1.20.2.66
@@ -359,7 +359,8 @@
 gchar *get_tmp_file			(void);
 const gchar *get_domain_name		(void);
 #ifdef G_OS_WIN32
-const gchar *get_cert_file(void);
+const gchar *get_themes_dir             (void);
+const gchar *get_cert_file		(void);
 #endif
 /* file / directory handling */
 off_t get_file_size		(const gchar	*file);

