#! /bin/sh
patch -p0 -f $* < $0
exit $?

Index: src/messageview.c
===================================================================
RCS file: /srv/cvs/claws-mail/claws/src/messageview.c,v
retrieving revision 1.94.2.194
retrieving revision 1.94.2.195
diff -u -r1.94.2.194 -r1.94.2.195
--- src/messageview.c	9 Dec 2008 16:59:36 -0000	1.94.2.194
+++ src/messageview.c	9 Jan 2009 08:09:21 -0000	1.94.2.195
@@ -674,6 +674,9 @@
 
 	gtk_widget_set_size_request(window, prefs_common.msgwin_width,
 				    prefs_common.msgwin_height);
+#ifdef G_OS_WIN32
+	gtk_window_move(GTK_WINDOW(window), 48, 48);
+#endif
 
 	msgview = messageview_create(mainwin);
 

