#! /bin/sh
patch -p0 -l -f $* < $0
exit $?

--- CMakeLists.txt.orig 2019-02-18 17:01:11.892790831 +0100
+++ CMakeLists.txt      2019-02-18 17:05:04.381951901 +0100
@@ -44,7 +44,7 @@
 set(REQUIRED_QT_VERSION 5.15.2)
 find_package(Qt${QT_MAJOR_VERSION} ${REQUIRED_QT_VERSION} CONFIG REQUIRED Widgets Xml Network PrintSupport)
 # On non-Apple UNIX DBus is required, on Apple Unix & Windows optional and it does not exist on Android
-if(UNIX AND NOT APPLE)
-     if(NOT ANDROID)
+if(WIN32 OR (UNIX AND NOT APPLE))
+     if(NOT WIN32 AND NOT ANDROID)
         find_package(Qt${QT_MAJOR_VERSION} ${REQUIRED_QT_VERSION} CONFIG REQUIRED DBus)
     endif()
